docs/performance/transformers-optimization.md

```diff
@@ -0,0 +1,256 @@
+```
+cd onnxruntime/python/tools/transformers/models/longformer/torch_extensions
+python setup.py install
+```
+It will generate a PyTorch extension file like "build/lib.linux-x86_64-3.8/longformer_attention.cpython-38-x86_64-linux-gnu.so" under the directory.
+
+Finally, convert longformer model to ONNX model like the following:
+```
+cd ..
+python convert_to_onnx.py -m longformer-base-4096
+```
+
+The exported ONNX model can only run on GPU right now.
+
+## Model Optimizer
+
+In your Python code, you can use the optimizer like the following:
+
+```python
+from onnxruntime.transformers import optimizer
+optimized_model = optimizer.optimize_model("bert.onnx", model_type='bert', num_heads=12, hidden_size=768)
+optimized_model.convert_float_to_float16()
+optimized_model.save_model_to_file("bert_fp16.onnx")
+```
+
+You can also use command line. Example of optimizing a BERT-large model to use mixed precision (float16):
+```console
+python -m onnxruntime.transformers.optimizer --input bert_large.onnx --output bert_large_fp16.onnx --num_heads 16 --hidden_size 1024 --float16
+```
+
+You can also download the latest script files from [here](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/). Then run it like the following:
+```console
+python optimizer.py --input bert.onnx --output bert_opt.onnx --model_type bert
+```
+
+### Optimizer Options
+
+
+- **input**: input model path
+- **output**: output model path
+- **model_type**: (*defaul: bert*)
+    There are 4 model types: *bert* (exported by PyTorch), *gpt2* (exported by PyTorch), and *bert_tf* (BERT exported by tf2onnx), *bert_keras* (BERT exported by keras2onnx) respectively.
+- **num_heads**: (*default: 12*)
+    Number of attention heads. BERT-base and BERT-large has 12 and 16 respectively.
+- **hidden_size**: (*default: 768*)
+    BERT-base and BERT-large has 768 and 1024 hidden nodes respectively.
+- **input_int32**: (*optional*)
+    Exported model ususally uses int64 tensor as input. If this flag is specified, int32 tensors will be used as input, and it could avoid un-necessary Cast nodes and get better performance.
+- **float16**: (*optional*)
+    By default, model uses float32 in computation. If this flag is specified, half-precision float will be used. This option is recommended for NVidia GPU with Tensor Core like V100 and T4. For older GPUs, float32 is likely faster.
+-  **use_gpu**: (*optional*)
+    When opt_level > 1, please set this flag for GPU inference.
+- **opt_level**: (*optional*)
+    Set a proper graph optimization level of OnnxRuntime: 0 - disable all (default), 1 - basic, 2 - extended, 99 - all. If the value is positive, OnnxRuntime will be used to optimize graph first.
+- **verbose**: (*optional*)
+    Print verbose information when this flag is specified.
+
+### Supported Models
```

Comment:
Maybe put this at the top, so that a user can decide whether the tool is relevant to them

Reply:
Done